### PR TITLE
Added positionOriginZ to TransformComponent

### PR DESCRIPTION
**Pull request by VasilisAndritsoudis**
From class ViewComponent:
    Changed:
    `viewRoot.translateZProperty().bind(entity.transformComponent.zProperty())`
    To:
    `viewRoot.translateZProperty().bind(entity.yProperty().subtract(entity.transformComponent.positionOriginZProperty()))`

From class TransformComponent:
  And added the necessary variable and function in order to work correctly:
  `private val propPositionOriginZ = SimpleDoubleProperty(0.0)`
  `fun positionOriginZProperty() = propPositionOriginZ`

Run some basic tests using the supplied 3D examples under fxgl-samples / ... / sandbox/test3d and everything seemed to be fine.

### CODE REVIEW

File: fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/TransformComponent.kt [hunk 1]
```diff
@@ -41,6 +41,7 @@ class TransformComponent(x: Double, y: Double, angle: Double, scaleX: Double, sc
 
     private val propPositionOriginX = SimpleDoubleProperty(0.0)
     private val propPositionOriginY = SimpleDoubleProperty(0.0)
+	private val propPositionOriginZ = SimpleDoubleProperty(0.0)
```
Comment (AlmasB) [inline fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/TransformComponent.kt:45]: Looks good, but please fix spacing. The IDE formatting should be set to *4 spaces, no Tabs*.

File: fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/TransformComponent.kt [hunk 2]
```diff
@@ -132,6 +133,7 @@ class TransformComponent(x: Double, y: Double, angle: Double, scaleX: Double, sc
 
     fun positionOriginXProperty() = propPositionOriginX
     fun positionOriginYProperty() = propPositionOriginY
+	fun positionOriginZProperty() = propPositionOriginZ
```
Comment (AlmasB) [inline fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/TransformComponent.kt:137]: Same comment as above

File: fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/ViewComponent.kt [hunk 3]
```diff
@@ -83,8 +83,7 @@ class ViewComponent : Component() {
     override fun onAdded() {
         viewRoot.translateXProperty().bind(entity.xProperty().subtract(entity.transformComponent.positionOriginXProperty()))
         viewRoot.translateYProperty().bind(entity.yProperty().subtract(entity.transformComponent.positionOriginYProperty()))
-
-        viewRoot.translateZProperty().bind(entity.transformComponent.zProperty())
+		viewRoot.translateZProperty().bind(entity.zProperty().subtract(entity.transformComponent.positionOriginZProperty()))
```
Comment (AlmasB) [inline fxgl-entity/src/main/kotlin/com/almasb/fxgl/entity/components/ViewComponent.kt:87]: Same comment

**Review by AlmasB — CHANGES_REQUESTED** [review]

**Comment by VasilisAndritsoudis** [comment]
@AlmasB I am not sure as to why the formatting was off. I opened the files in two different environments (with tab size set to 4 spaces) and they looked fine. However when uploaded to github there was an error with the spacing. I fixed them through github hoping it will work.

**Review by AlmasB — APPROVED** [review]
Looks good. ViewComponent still had a tab, which I replaced with 4 spaces, but TransformComponent was fine. Thanks.

**Comment by AlmasB** [comment]
Closes #923

**Pull request merged**